WcaOnRails/app/assets/javascripts/application.js

```diff
@@ -46,25 +46,38 @@ wca.cancelPendingAjaxAndAjax = function(id, options) {
   return wca._pendingAjaxById[id];
 };
 
+function goodDate(element) {
+  element.parent().siblings('p').removeClass('alert alert-danger');
+  element.removeClass('alert-danger');
+}
+
+function badDate(element) {
+  element.parent().siblings('p').addClass('alert alert-danger');
+  element.parent().siblings('p').fadeIn(200).fadeOut(200).fadeIn(200).fadeOut(200).fadeIn(200);
+  element.addClass('alert-danger');
+}
+
 $(function() {
   $('.dropdown-toggle').dropdownHover();
   $('form.are-you-sure').areYouSure();
 
   // Copied (and modified by jfly) from
-  //  https://github.com/zpaulovics/datetimepicker-rails
+  // https://github.com/zpaulovics/datetimepicker-rails
   $('.date_picker.form-control, .datetime_picker.form-control').datetimepicker({
     useStrict: true, keepInvalid: true, useCurrent: false
   });
 
-  $('.date_picker.form-control, .datetime_picker.form-control').on('dp.error', function(e){
-    $(this).parent().siblings('p').addClass('alert alert-danger');
-    $(this).parent().siblings('p').fadeIn(200).fadeOut(200).fadeIn(200).fadeOut(200).fadeIn(200);
-    $(this).addClass('alert-danger');
-  });
+  $('.date_picker.form-control').on('blur', function(){
+    var $this = $(this);
+    var datetimepicker = $this.data("DateTimePicker");
+    var val = $this.val();
+    var valid = moment(val, datetimepicker.format(), true).isValid();
 
-  $('.date_picker.form-control, .datetime_picker.form-control').on('dp.change', function(e){
-    $(this).parent().siblings('p').removeClass('alert alert-danger');
-    $(this).removeClass('alert-danger');
+    if(valid) {
+      goodDate($this);
```

Comment:
There could be, if we make them more general methods, like bad Input and
goodInput and use them for other stuff.

But if that's not in the long-term plans, we could move them inside the ifs.
Em 08/02/2016 17:32, "Jeremy Fleischman" <notifications@github.com>
escreveu:

> In WcaOnRails/app/assets/javascripts/application.js
> <https://github.com/cubing/worldcubeassociation.org/pull/379#discussion_r52195948>
> :
>
> >
> > -  $('.date_picker.form-control, .datetime_picker.form-control').on('dp.change', function(e){
> > -    $(this).parent().siblings('p').removeClass('alert alert-danger');
> > -    $(this).removeClass('alert-danger');
> > +    if(valid) {
> > +      goodDate($this);
>
> Is there any value in making goodDate and badDate separate functions? I
> think it would make the code easier to read if you just moved them inside
> this function.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cubing/worldcubeassociation.org/pull/379/files#r52195948>
> .
>